Bugfix9

Kredcool opened the pull request:
this attempts to fix the not being able to go over 360

Kredcool committed:
this somehow breaks the driving code

KenwoodFox commented on dashboard/.python-version:
```diff
@@ -0,0 +1 @@
+3.8.0
```
Dont include this

KenwoodFox commented on docs/.python-version:
```diff
@@ -0,0 +1 @@
+3.8.0
```
Dont include this

KenwoodFox commented on rio/subsystems/drivetrain.py:
```diff
@@ -105,68 +120,141 @@ def configureNetworkTables(self):
+                "steer"
+            ]["ratio"]
+
+            # this checks if the desiredState is more than 349 and currentRef is 59 then adds 2 pi to currentRef
```
Degrees and radians dont mix

KenwoodFox commented on rio/subsystems/drivetrain.py:
```diff
@@ -105,68 +120,141 @@ def configureNetworkTables(self):
+            ]["ratio"]
+
+            # this checks if the desiredState is more than 349 and currentRef is 59 then adds 2 pi to currentRef
+            if (
```
This is pretty messy

KenwoodFox commented on rio/subsystems/drivetrain.py:
```diff
@@ -105,68 +120,141 @@ def configureNetworkTables(self):
+                and self.desiredState.angle.radians() >= 6.3490912
+                and currentRef.angle.radians() <= 1.0297
+            ):
+                currentRef = currentRef.angle.radians() + math.pi + math.pi
```
This is an odd way of adding 2pi but i see what you're trying to do

Kredcool replied:
```python
2 * math.pi```

KenwoodFox committed:
Sperate simulator, and irl controls

KenwoodFox committed:
Dont inclide .python-version from pyenv

KenwoodFox committed:
This is slightly buggy (looses track a lot) but it describes the sum pretty well

KenwoodFox committed:
Bump pipfile

KenwoodFox committed:
Clean up new sumation math

KenwoodFox committed:
This does work but...
Few major bugs, one, im not sure if the setConversionFactor() is working
Two, setPosition(0) does not seem to reset the encoder to 0 on restart so, thats a bug
Three, piping optimizedState into the new sum system seems to break it, and idk why
